OpenRA.Mods.Common/Lint/CheckPalettes.cs

```diff
@@ -29,14 +32,36 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset ru
 					var fields = traitInfo.GetType().GetFields();
 					foreach (var field in fields.Where(x => x.HasAttribute<PaletteReferenceAttribute>()))
 					{
+						var isPlayerPalette = false;
+
+						var paletteReference = field.GetCustomAttributes<PaletteReferenceAttribute>(true).FirstOrDefault();
+						if (paletteReference != null)
+						{
+							if (!string.IsNullOrEmpty(paletteReference.PlayerPaletteReferenceSwitch))
+							{
+								var fieldInfo = fields.FirstOrDefault(f => f.Name == paletteReference.PlayerPaletteReferenceSwitch);
```

Comment:
This should probably also be `.First()`, just to get a slightly better exception if it crashes (with or without the `OrDefault` part it will crash if there is no such field; that is good because it will indicate a major error in the trait's Info class that needs to be addressed).